packages/manager/src/views/WebPropertyDetailPage/components/Dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,346 @@
+import { pageLinks } from '@app/links';
+import {
+  useGetHalfYearlyDeploymentsTime,
+  useGetMonthlyDeploymentChart,
+  useGetMonthlyDeploymentChartWithEphemeral,
+  useGetMonthlyDeploymentsTime,
+  useGetQuarterlyDeploymentsTime,
+  useGetTotalDeployments,
+  useGetTotalDeploymentsForApps,
+  useGetYearlyDeploymentsTime
+} from '@app/services/analytics';
+import { useGetSPAPropGroupByName } from '@app/services/spaProperty';
+import {
+  Chart,
+  ChartAxis,
+  ChartDonut,
+  ChartGroup,
+  ChartLine,
+  ChartThemeColor,
+  ChartVoronoiContainer
+} from '@patternfly/react-charts';
+import {
+  Card,
+  CardBody,
+  CardHeader,
+  CardTitle,
+  EmptyState,
+  EmptyStateIcon,
+  Grid,
+  GridItem,
+  Skeleton,
+  Split,
+  SplitItem,
+  Title
+} from '@patternfly/react-core';
+import { CubesIcon } from '@patternfly/react-icons';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useMemo } from 'react';
+import toast from 'react-hot-toast';
+
+const TotalDeploymentCardFields = ['Dev', 'QA', 'Stage', 'Prod'];
+const DeploymentTimeFrames = ['month', 'quarter', 'half year', 'year'];
+type IGraphData = {
+  name: string;
+  x: string;
+  y: number;
+};
+type ITotalMonthlyDeploymentData = {
+  dev: IGraphData[] | undefined;
```

Comment:
Use `dev?: IGraphData[]` instead of an undefine union

Reply:
Done